src/mlb_managers.py

```diff
@@ -0,0 +1,278 @@
+                if not self._no_data_warning_logged:
+                    self.logger.warning("No recent MLB games found")
+                    self._no_data_warning_logged = True
+                return False
+            
+            # Filter for recent games (last 24 hours)
+            now = datetime.now(pytz.UTC)
+            recent_games = [game for game in recent_games 
+                           if game.get('is_final', False) and 
+                           game.get('start_time_utc') and 
+                           (now - game['start_time_utc']).total_seconds() < 86400]
+            
+            if not recent_games:
+                self.logger.debug("No recent MLB games in last 24 hours")
+                return False
+            
+            # Display each game
+            for game in recent_games:
+                self._display_single_game(game)
+                time.sleep(2)
+            
+            return True
+            
+        except Exception as e:
+            self.logger.error(f"Error displaying recent MLB games: {e}")
+            return False
+
+
+class MLBUpcomingManager(BaseMLBManager):
+    """Manager for upcoming MLB games."""
+    
+    def __init__(self, config: Dict[str, Any], display_manager: DisplayManager, cache_manager: CacheManager):
+        super().__init__(config, display_manager, cache_manager)
+        self.logger.info("MLB Upcoming Manager initialized")
+
+    def get_duration(self) -> int:
+        """Get display duration for upcoming MLB games."""
+        return self.mode_config.get("duration", 6)
+
+    def display(self) -> bool:
+        """Display upcoming MLB games."""
+        try:
+            # Fetch upcoming games using the new architecture
+            upcoming_games = self._get_partial_schedule_data(datetime.now().year)
+            
+            if not upcoming_games:
+                if not self._no_data_warning_logged:
+                    self.logger.warning("No upcoming MLB games found")
+                    self._no_data_warning_logged = True
+                return False
+            
+            # Filter for upcoming games (next 7 days)
+            now = datetime.now(pytz.UTC)
+            upcoming_games = [game for game in upcoming_games 
+                             if game.get('is_upcoming', False) and 
+                             game.get('start_time_utc') and 
+                             (game['start_time_utc'] - now).total_seconds() < 604800]
+            
+            if not upcoming_games:
+                self.logger.debug("No upcoming MLB games in next 7 days")
+                return False
+            
+            # Display each game
+            for game in upcoming_games:
+                self._display_single_game(game)
+                time.sleep(2)
+            
+            return True
+            
+        except Exception as e:
+            self.logger.error(f"Error displaying upcoming MLB games: {e}")
+            return False
```

Comment:
YOu should be able to simplify these with just extending `SportsRecent` and `SportsUpcoming`